lightkurve/tests/test_io.py

```diff
@@ -0,0 +1,92 @@
+import os
+
+from astropy.io import fits
+from numpy.testing import assert_array_equal
+import pytest
+
+from .. import PACKAGEDIR, KeplerTargetPixelFile, TessTargetPixelFile
+from ..io import detect_filetype, read, read_k2sff
+from ..search import search_lightcurvefile
+
+
+def test_detect_filetype():
+    """Can we detect the correct filetype?"""
+    k2_path = os.path.join(PACKAGEDIR, "tests", "data", "test-tpf-star.fits")
+    tess_path = os.path.join(PACKAGEDIR, "tests", "data", "tess25155310-s01-first-cadences.fits.gz")
+    assert detect_filetype(fits.open(k2_path)) == 'KeplerTargetPixelFile'
+    assert detect_filetype(fits.open(tess_path)) == 'TessTargetPixelFile'
+
+
+def test_read():
+    # define paths to k2 and  tess data
+    k2_path = os.path.join(PACKAGEDIR, "tests", "data", "test-tpf-star.fits")
+    tess_path = os.path.join(PACKAGEDIR, "tests", "data", "tess25155310-s01-first-cadences.fits.gz")
+    # Ensure files are read in as the correct object
+    k2tpf = read(k2_path)
+    assert(isinstance(k2tpf, KeplerTargetPixelFile))
+    tesstpf = read(tess_path)
+    assert(isinstance(tesstpf, TessTargetPixelFile))
+    # Open should fail if the filetype is not recognized
+    try:
+        read(os.path.join(PACKAGEDIR, "data", "lightkurve.mplstyle"))
+    except (ValueError, IOError):
+        pass
+    # Can you instantiate with a path?
+    assert(isinstance(KeplerTargetPixelFile(k2_path), KeplerTargetPixelFile))
+    assert(isinstance(TessTargetPixelFile(tess_path), TessTargetPixelFile))
+    # Can open take a quality_bitmask argument?
+    assert(read(k2_path, quality_bitmask='hard').quality_bitmask == 'hard')
+
+
+def test_filenotfound():
+    """Regression test for #540; ensure lk.read() yields `FileNotFoundError`."""
+    # Python 2 uses IOError instead of FileNotFoundError;
+    # the block below can be removed when we drop Python 2 support.
+    try:
+        FileNotFoundError
+    except NameError:
+        FileNotFoundError = IOError
+    with pytest.raises(FileNotFoundError):
+        read("DOESNOTEXIST")
+
+
+@pytest.mark.remote_data
+def test_read_k2sff():
```

Comment:
Do you want to save a shortened K2sff file for tests? Same with EVEREST?

Reply:
I'm going to use remote data for now, until the API settles down.